dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClasspathChecker.java

```diff
@@ -49,33 +49,43 @@
 
   private static final Logger logger = Logger.getLogger(ClasspathChecker.class.getName());
 
-  public static ClasspathChecker create(
-      boolean onlyReachable, List<Path> jarPaths, Iterable<Path> entryPoints)
+  public static ClasspathChecker create(List<Path> jarPaths, Iterable<Path> entryPoints)
       throws IOException {
     Preconditions.checkArgument(
         !jarPaths.isEmpty(),
         "The linkage classpath is empty. Specify input to supply one or more jar files");
     ClassDumper dumper = ClassDumper.create(jarPaths);
-    return new ClasspathChecker(onlyReachable, dumper, entryPoints);
-  }
 
-  /**
-   * If true, the report excludes linkage errors on classes that are not reachable
-   * from the entry points of the class usage graph.
-   */
-  private final boolean reportOnlyReachable;
+    ImmutableMap.Builder<Path, SymbolReferenceSet> jarToSymbolBuilder = ImmutableMap.builder();
+    for (Path jarPath : jarPaths) {
+      jarToSymbolBuilder.put(jarPath, ClassDumper.scanSymbolReferencesInJar(jarPath));
+    }
+    ImmutableMap<Path, SymbolReferenceSet> jarToSymbols = jarToSymbolBuilder.build();
+
+    ImmutableSet.Builder<ClassSymbolReference> classReferenceBuilder = ImmutableSet.builder();
+    for (SymbolReferenceSet symbolReferenceSet : jarToSymbols.values()) {
+      classReferenceBuilder.addAll(symbolReferenceSet.getClassReferences());
+    }
+
+    ClassReferenceGraph classReferenceGraph =
+        ClassReferenceGraph.create(classReferenceBuilder.build(), ImmutableSet.copyOf(entryPoints));
+
+    return new ClasspathChecker(dumper, jarToSymbols, classReferenceGraph);
+  }
 
   private final ClassDumper classDumper;
 
-  private final ImmutableSet<Path> entryPoints;
-  
+  private final ImmutableMap<Path, SymbolReferenceSet> jarToSymbols;
+
```

Comment:
You don't really need these blank lines in between simple field declarations, do you? Up to you.

Reply:
Removed.